Microsoft.Toolkit.Uwp.UI.Controls/BladeView/BladeView.cs

```diff
@@ -25,12 +28,17 @@ public partial class BladeView : ItemsControl
     {
         private ScrollViewer _scrollViewer;
 
+        private Dictionary<BladeItem, Size> _cachedBladeItemSizes = new Dictionary<BladeItem, Size>();
+
         /// <summary>
         /// Initializes a new instance of the <see cref="BladeView"/> class.
         /// </summary>
         public BladeView()
         {
             DefaultStyleKey = typeof(BladeView);
```

Comment:
We should listen to the items changing to make sure new items have width and height set when in full mode. You can subscribe to the `Items.VectorChanged` event. You'll need to check that they already exist in the dictionary (from within the event handler) because the event will fire multiple times.